.github/workflows/dekaf.yaml

```diff
@@ -6,6 +6,16 @@ on:
     paths:
       - "crates/dekaf/**"
       - "crates/proto-**"
+      - "crates/allocator"
```

Comment:
I think I'm coming around to the idea that Dekaf should just be another binary in the flow image, as opposed to its own image. But for the moment, this just has it rebuild when any of its local crate deps change